modyn/selector/internal/selector_strategies/abstract_downsample_strategy.py

```diff
@@ -23,11 +23,36 @@ class AbstractDownsampleStrategy(AbstractPresampleStrategy):
     def __init__(self, config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int):
         super().__init__(config, modyn_config, pipeline_id, maximum_keys_in_memory)
 
-        if "downsampled_batch_size" not in self._config:
-            raise ValueError("To use downsampling strategies, you have to specify the downsampled_batch_size")
-        self.downsampled_batch_size = self._config["downsampled_batch_size"]
-
-        if not isinstance(self.downsampled_batch_size, int):
-            raise ValueError("The downsampled batch size must be an integer")
+        if "sample_before_batch" not in config:
+            raise ValueError(
+                "Please specify if you want to sample and then batch or vice versa. "
+                "Use the sample_before_batch parameter"
+            )
+        self.sample_before_batch = config["sample_before_batch"]
+
+        if self.sample_before_batch and "downsampling_period" in config:
+            raise ValueError("downsampling_period can be used only in sample-then-batch.")
+        self.downsampling_period = config.get("downsampling_period", 1)
+
+        if self.sample_before_batch:
```

Comment:
could we maybe name the field `sample_then_batch` everywhere for consistency?

Reply:
Yes